lib/plugin/instance.rb

```diff
@@ -418,7 +418,11 @@ def register_custom_html(hash)
   end
 
   def register_html_builder(name, &block)
-    DiscoursePluginRegistry.register_html_builder(name, &block)
+    reloadable_patch do |plugin|
+      if plugin.enabled?
```

Comment:
I don't think this will work as you expect. It will only check `plugin.enabled?` when the app boots, and will only check the `default` database in a multisite cluster. The solution is to do something like [this](https://github.com/discourse/discourse/blob/master/app/models/user.rb#L267-L289). Store a reference to the plugin for each html_builder, and then check `plugin.enabled?` at runtime.

There are more details about this problem in [this commit message](https://github.com/discourse/discourse/commit/0d0d78841b02c9747796b6aa6cfe58c0d6eec457#diff-b1e56232326e1a7a433641f2c21261bd).

Reply:
I agree the whole situation is confusing, but which API is confusing? `register` happens when the app boots, so it is impossible for it to update the site setting live.

`reloadable_patch` is used for auto-reloading in development, it does nothing in production, and has nothing to do with multisite environments.